core/modules/config.js

```diff
@@ -38,4 +38,18 @@ exports.htmlBlockElements = "address,article,aside,audio,blockquote,canvas,dd,de
 
 exports.htmlUnsafeElements = "script".split(",");
 
+// Character code lookups used in charcode operator and macro
+exports.lookup = {
+	"tab": 9,
+	"\\t": 9,
+	"lf": 10,
+	"\\n": 10,
+	"ff": 12,
+	"\\f": 12,
+	"cr": 13,
+	"\\r": 13,
+	"\\_": 32,
```

Comment:
Why replace `\_` with a space? It doesn't seem to be a standard replacement.

And why are we only converting some of the JS character codes? Why not support all of the [standard ones](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String#escape_sequences)?

Reply:
```
\0 | null character (U+0000 NULL)
```

I think the null-char will be **problematic** if saved **in its binary form** in eg: .TID files and opened with other editors. It's not really defined, what they do with that character. In the best case they will filter it out and remove it. But eg: notepad++ [will do weird things](https://stackoverflow.com/questions/58689285/why-does-the-notepad-null-character-not-paste).
 
```
\' | single quote (U+0027 APOSTROPHE)
\" | double quote (U+0022 QUOTATION MARK)
\\ | backslash (U+005C REVERSE SOLIDUS)
```

I thought. there is no real need to implement them, our users already know, that they can cover single quotes inside double quotes and vice versa. eg. `" ' "` and `' " '` or `""" " """` which IMO is an equivalent to `"\""`

I don't have a problem to implement the escaped quotes and backslash, but then we probably will need to implement them globally for **all operators, string functions, macros and widgets**. If that's intended, it's far beyond the intention of this PR.

```
\v | vertical tab (U+000B LINE TABULATION)
\b | backspace (U+0008 BACKSPACE)
```

The vertical tab imo makes no sense. I can't think of any usecase other than controlling printer behaviour. But since printers are not really controlled that way anymore, I think it would create a completely unpredictable behaviour.

BACKSPACE is a control character that will remove the last character. It can be losslessly substituted by not creating that character at all. So I don't really see a usecase for it.

Reply:
>Why replace \_ with a space? It doesn't seem to be a standard replacement.

I think it would be useful, because it's more visible than ` ` if used with string manipulation.

Reply:
@pmario 

> The vertical tab imo makes no sense. I can't think of any usecase other than controlling printer behaviour. But since printers are not really controlled that way anymore, I think it would create a completely unpredictable behaviour.

I distrust the logic here. If I've learned anything it is that I can't predict how people use the software that I build. It seems really weird to pick and choose which character codes we support.

> > Why replace _ with a space? It doesn't seem to be a standard replacement.
> 
> I think it would be useful, because it's more visible than ` ` <- there is a space if used with string manipulation.

It just means that we end up with a strange mix of standard and non-standard conversions, which is all highly opinionated.